src/main/java/org/springframework/data/repository/config/RepositoryConfigurationExtensionSupport.java

```diff
@@ -106,7 +109,12 @@ public String getDefaultNamedQueryLocation() {
 	}
 
 	public void registerBeansForRoot(BeanDefinitionRegistry registry,
-			RepositoryConfigurationSource configurationSource) {}
+			RepositoryConfigurationSource configurationSource) {
+
+		// A proxy RepositoryMethodContext for dependency injection
+		registry.registerBeanDefinition("repositoryMethodContextFactory",
```

Comment:
Likely we need to check whether there is already such a bean in case an application uses multiple Spring Data modules or a `@Enable…Repositories` is used twice.

Reply:
Updated.